src/plugins/intel_cpu/src/cpu_memory.h

```diff
@@ -116,6 +116,21 @@ class DnnlMemoryMngr : public IMemoryMngrObserver {
     std::unique_ptr<IMemoryMngr> _pMemMngr;
 };
 
+class LockBasedMemoryMngr : public IMemoryMngrObserver {
```

Comment:
@maxnick May I ask why this LockBasedMemoryMngr is invented? Sounds it is only used in input node when it is a constant node.

Reply:
Yes exactly. This memory mngr type is necessary for handling concurrent access from different streams. The concurrent access happens due to sharing the constants among streams.